sn_node/src/put_validation.rs

```diff
@@ -476,18 +473,42 @@ impl Node {
             .reward_wallet_balance
             .set(wallet.balance().as_nano() as i64);
 
-        // check payment is sufficient
-        let current_store_cost =
-            self.network.get_local_storecost().await.map_err(|e| {
-                ProtocolError::RecordNotStored(pretty_key.clone(), format!("{e:?}"))
-            })?;
+        if royalties_cash_notes.is_empty() {
```

Comment:
This fail we should put _after_ we grab any money sent to us. Just in case we were sent something valid.

Reply:
I think we are doing that, it's after we deposited any cash notes belonging to us